auction-server/src/opportunity/service/estimate_price.rs

```diff
@@ -0,0 +1,22 @@
+use {
+    super::{
+        ChainTypeSvm,
+        Service,
+    },
+    crate::{
+        api::RestError,
+        opportunity::entities,
+    },
+};
+
+pub struct EstimatePriceInput {
+    pub quote_create: entities::QuoteCreate,
+}
+
+impl Service<ChainTypeSvm> {
+    #[tracing::instrument(skip_all)]
+    pub async fn estimate_price(&self, _input: EstimatePriceInput) -> Result<u64, RestError> {
+        // TODO implement
+        return Ok(0);
```

Comment:
i think i understand, you intend for the server to grab the reference price? there are some tradeoffs here:
- if the server grabs the reference price, then all the load is at the server level. e.g. if this process is very long/slow, then the server can get backed up in api calls here
- if the wallet provides a few prices/the best price, then this method involves a lot less work and fewer external deps.

we should at the least think about some load management here to balance making any external calls here